### PR TITLE
add sha512 as an optional checksum

### PR DESCRIPTION
**Pull request by elmiko**
This change brings in the `sha512` as a possible key for checksum values
associated with the `artifact` directive. It also updates the relevent
tests and documentation.

closes issue #471

### CODE REVIEW

File: tests/test_validate_cache.py [hunk 1]
```diff
@@ -119,9 +120,9 @@ def test_cekit_cannot_add_artifact_without_checksum(tmpdir):
     result = run_cekit_cache(['--work-dir',
                               work_dir,
                               'add',
-                              artifact], 2)
+                              artifact], 1)
 
-    assert "At least one checksum must be provided" in result.output
+    assert "Cannot cache artifact without checksum" in result.output
```
Comment (elmiko) [inline tests/test_validate_cache.py]: changing this test was the only thing that confused me. it seemed to be working before, but after my changes it broke and i'm not sure why. fixing it was just a matter of changing the expected result code and the error text.

File: cekit/cache/cli.py [hunk 2]
```diff
@@ -36,11 +36,12 @@ def ls():
 @click.option('--md5', metavar="CHECKSUM", help="The md5 checksum of the artifact.")
 @click.option('--sha1', metavar="CHECKSUM", help="The sha1 checksum of the artifact.")
 @click.option('--sha256', metavar="CHECKSUM", help="The sha256 checksum of the artifact.")
-def add(location, md5, sha1, sha256):  # pylint: disable=unused-argument
-    if not (md5 or sha1 or sha256):
+@click.option('--sha512', metavar="CHECKSUM", help="The sha512 checksum of the artifact.")
+def add(location, md5, sha1, sha256, sha512):  # pylint: disable=unused-argument
+    if not (md5 or sha1 or sha256, sha512):
```
Comment (goldmann) [inline cekit/cache/cli.py]: Here is the issue, there should be no comma, but `or` string.

File: tests/test_validate_cache.py (hunk 1 above)
Comment (goldmann) [reply tests/test_validate_cache.py]: Added a comment above that points where the issue is.

File: cekit/cache/cli.py (hunk 2 above)
Comment (elmiko) [reply cekit/cache/cli.py]: oops, moving too fast. good spot!

**Comment by elmiko** [comment]
ok, added the `or` and reverted the other test.

thanks again for the help @goldmann =)

**Comment by goldmann** [comment]
This is why we have reviews :)

**Comment by goldmann** [comment]
@elmiko Thank you!

**Comment by elmiko** [comment]
glad to help!

**Pull request merged**